test/ttmlir/Dialect/TTNN/sharding_matmul_override_32.mlir

```diff
@@ -0,0 +1,14 @@
+// RUN: ttmlir-opt --ttir-to-ttnn-backend-pipeline="enable-optimizer=true sharding-pass-enabled=true max-legal-layouts=32" %s | FileCheck %s
+#any_device_tile = #tt.operand_constraint<dram|l1|tile|any_device_tile>
+module attributes {} {
+  func.func @forward(%arg0: tensor<64x128xbf16>, %arg1: tensor<128x96xbf16>, %arg2: tensor<96x64xbf16>) -> tensor<64x64xbf16> {
+    // CHECK: #[[L1_:.*]] = #tt.memory_space<l1>
+    // CHECK: #[[LAYOUT_7:.*]] = #tt.layout<(d0, d1) -> (d0, d1), undef, <2x3>, memref<1x1x!tt.tile<32x32, bf16>, #l1_>, block_sharded>
```

Comment:
This is a bit weird way to check that max legal layouts is 32. Because its 32 output of matmul here will be block_sharded? Optimizer decision might change causing this test to be flaky in the future and bit pointless?

Reply:
Yes, makes sense. The idea was to test that anything at all is sharded, compared with the case when max legal layouts is 0. We could match for something like `.*_sharded` or not test for this at all?

Reply:
Yeah let's make regex in such a way to match all types of sharding.